File: Projekt3/backend/app/schemas/room.py
from datetime import datetime

from pydantic import BaseModel


class RoomOut(BaseModel):
    id: int
    name: str
    capacity: int
    description: str

    model_config = {"from_attributes": True}


class RoomCreate(BaseModel):
    name: str
    capacity: int
    description: str = ""


class RoomUpdate(BaseModel):
    name: str | None = None
    capacity: int | None = None
    description: str | None = None


class BookingOut(BaseModel):
    id: int
    room_id: int
    user_id: int
    start_time: datetime
    end_time: datetime
    title: str | None = None

    model_config = {"from_attributes": True}


class BookingCreate(BaseModel):
    room_id: int
    start_time: datetime
    end_time: datetime
    title: str | None = None

File: Projekt3/frontend/src/types/Room.ts
export interface Room {
  id: number;
  name: string;
  capacity: number;
  description: string;
}

export interface Booking {
  id: number;
  room_id: number;
  user_id: number;
  title: string;
  start_time: string;
  end_time: string;
}

export interface RoomCreate {
  name: string;
  capacity: number;
  description: string;
}

export interface RoomUpdate {
  name?: string;
  capacity?: number;
  description?: string;
}

export interface BookingCreate {
  room_id: number;
  title: string;
  start_time: string;
  end_time: string;
}
